### PR TITLE
gnrc/ipv6_auto_subnets: add CONFIG_GNRC_IPV6_AUTO_SUBNETS_STATIC option

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

In situations with high packet loss, if all sync packets are lost, conflicting subnets will be configured.

If the network is static, this can be prevented by always using the highest number of subnets that has been observed.

This assumes no nodes are physically added / removed from the network. This is typically given in a wired sensor network.

### Testing procedure

Add 

    CFLAGS += -DCONFIG_GNRC_IPV6_AUTO_SUBNETS_STATIC=1

to the `Makefile` of `examples/gnrc_networking_subnets`.

Configure a topology with multiple routers on the same subnet. Now even if you remove a router, the subnet assignments should not change.

### CODE REVIEW

**Comment by benpicco** [comment]
For that I would first need to add Kconfig to that subsystem

File: sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c [hunk 1]
```diff
@@ -622,6 +634,10 @@ static void *_eventloop(void *arg)
     uint8_t idx_start = 0;
     uint8_t subnets = local_subnets;
     uint8_t tx_period = CONFIG_GNRC_IPV6_AUTO_SUBNETS_TX_PER_PERIOD;
+#if CONFIG_GNRC_IPV6_AUTO_SUBNETS_STATIC
+    uint8_t _idx_old = 0;
+    uint8_t _subnets_old = 0;
```
Comment (miri64) [inline sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c:639]: The `_` prefix usually is used for static globals (copying somewhat the python paradigm to mark private members as such). So why is it used here?

File: sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c [hunk 2]
```diff
@@ -648,6 +664,18 @@ static void *_eventloop(void *arg)
                 /* send subnet announcement */
                 _send_announce(local_subnets, &timeout_timer, &timeout_msg);
             } else {
+#if CONFIG_GNRC_IPV6_AUTO_SUBNETS_STATIC
```
Comment (miri64) [inline sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c:667]: If you used
```suggestion
                if (IS_ACTIVE(CONFIG_GNRC_IPV6_AUTO_SUBNETS_STATIC)) {
```
here instead, would this result in the same binary? Would have the advantage, that this part would be at least compile tested, whenever `gnrc_ipv6_auto_subnets` is included.

**Review by miri64 — CHANGES_REQUESTED** [review]

File: sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c (hunk 2 above)
Comment (miri64) [reply sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c:667]: (you need to remove the parenthesis around `0` in the `#define` of `CONFIG_GNRC_IPV6_AUTO_SUBNETS_STATIC` for this to be compilable in the first place)

Comment (benpicco) [reply sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c:667]: No need for `IS_ACTIVE` if it's default-defined to 0

Comment (benpicco) [reply sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c]: I can confirm that the same binary is generated.

Comment (miri64) [reply sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c]: `IS_ACTIVE` has the added benefit that if it is also 0, when the macro is not defined (when for whatever reason in the future >Kconfig :eyes:< the macro is not defined).

**Review by miri64 — APPROVED** [review]
ACK

**Pull request merged**